Pixel Game/Assets/Scripts/Damage.cs

```diff
@@ -5,47 +5,63 @@
 
 public class Damage : MonoBehaviour
 {
-    public enum Group { Players, Mobs }
-
     [SerializeField]
-    private int attackPower = 0;
+    private bool canBeAttacked;
     [SerializeField]
-    private float attackCooldown = 1;
+    private bool canAttack;
     [SerializeField]
-    private Group group;
+    private float attackCooldown = 1;
     [SerializeField]
-    private GameObject damageReceiver;
-    
+    private GameObject owner;
     private DateTime lastAttackTime;
     private Health health;
-
-    private bool CanInflictDamage => attackPower > 0;
-    private bool CanTakeDamage => damageReceiver != null;
-
+    private Attack Attack
+    {
+        get
+        {
+            var attack = owner.GetComponent<Attack>();
+            if (attack == null)
+                attack = owner.GetComponent<Weapon>().Owner.GetComponent<Attack>();
+            return attack;
+        }
+    }
+    
     private void Start()
     {
-        if (damageReceiver != null)
-            health = damageReceiver.GetComponent<Health>();
+        SetOwner(owner);
     }
+
+    public void SetOwner(GameObject owner)
```

Comment:
Is this owner set in the inspector?

Reply:
Yes, except when it comes to projectiles. To enable use of a projectile, its prefab needs to be dragged to a Weapon component. When this is done, the code takes care of setting the weapon as owner.